### PR TITLE
Move test utils sections to actions and components

### PR DESCRIPTION
**Pull request by redonkulus**
@mridgway

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

File: docs/api/Actions.md [hunk 1]
```diff
@@ -57,3 +57,79 @@ Executes another action and allows waiting for the `done` callback to be called.
...
+
+When testing your actions, you can use our `MockActionContext` library and pass an instance to your action to record the methods that the action calls on the context.
+
+When `dispatch` is called, it will push an object to the `dispatchCalls` array. Each object contains a `name` and `payload` key.
+
+When `executeAction` is called, it will push an object to the `executeActionCalls` array. Each object contains an `action` and `payload` key.
+
+`getStore` calls will be proxied to a dispatcher instance, which you can register stores to via `MockActionContext.registerStore(MockStore)`.
+
+#### Usage
```
Comment (mridgway) [inline docs/api/Actions.md]: Too many '#'

**Comment by mridgway** [comment]
:+1: with 1 nitpick

File: docs/api/Actions.md (hunk 1 above)
Comment (redonkulus) [inline docs/api/Actions.md]: good catch updated.

**Pull request merged**